### ISSUE TITLE
[App] Bug: scalar chart not finding logged values

### ISSUE THREAD

**Issue by andreimargeloiu** [wandb/wandb#3074]
**Describe the bug**
I'm trying to create a scalar chart from a logged value.

![CleanShot 2021-12-21 at 20 27 07](https://user-images.githubusercontent.com/18227298/146980253-d0b7bbbb-a0fe-4a5a-9433-adaada1f7fc1.png)

However, some scalar charts (e.g., the middle one from the top) seem not to find the corresponding values. 

![CleanShot 2021-12-21 at 20 27 42](https://user-images.githubusercontent.com/18227298/146980293-ac64d797-a344-4852-ab6a-2449e32e140e.png)

But the value is clearly logged, and visible in a simple scatter plot.

![CleanShot 2021-12-21 at 20 28 09](https://user-images.githubusercontent.com/18227298/146980323-c152dc89-7cc7-4d78-a543-068b2c2d1a8e.png)

**Expected behavior**
I would expect to see the value of the logged value, not an error.

**Operating System**
 - OS: [e.g. iOS] MacOS
 - Browser [e.g. chrome, safari] Microsoft Edge
     - wandb 0.12.9 
     - logs created from the wandb integration with pytorch-lightening 1.5.5
     - python 3.7.9

**Comment by laxels** [comment]
Hi there @andreimargeloiu! Sorry about this, we'll look into it straight away. Can you link us the workspace where you're seeing this issue?

**Comment by andreimargeloiu** [comment]
Run: https://wandb.ai/andreimargeloiu/low-data/runs/wvkt6mvz?workspace=user-andreimargeloiu

**Comment by andreimargeloiu** [comment]
Here I'm detailing my use case and asking how to use wandb for it.

I'm training a model in my codebase and choosing the best model based on the validation loss. However, in the wandb UI I want to see the accuracy of the best model (note that the accuracy of the best model chosen on the validation loss can be smaller than the model with the best accuracy).

I can't simply use wandb.define_metric(my_metric, summary='max'), because, commonly, the model with the best validation loss doesn't necessarily have the best validation accuracy.

*Current solution*: After training, compute the accuracy of the best model, and save using wandb.log, giving a chart with only one data point saved. Then, I create a scalar chart that simply shows the value of

Can you suggest how to simplify this workflow?

**Comment by laxels** [comment]
Hi there @andreimargeloiu! Sorry for the late response; just got caught up post-holidays.

Based on your workflow, I think you can do the following:

For every run, do
```
wandb.define_metric('validation_loss', summary='min')
wandb.define_metric('validation_accuracy', summary='max')
```

This configuration will tell W&B to keep track of the minimum `validation_loss` as well as the maximum `validation_accuracy` for every single model you're training. Then, in your project workspace (the one with the large table with all your training runs), you can sort your models by the lowest `validation_loss.min` and you will be able to see that model's `validation_accuracy.max` as well.

I may have misunderstood what you're trying to do, so please let me know if this doesn't solve your use case. Please also let me know if this explanation is unclear in any way; I can elaborate further.

I also looked into the issue that's causing your scalar chart to not display your `bestmodel_train/` metrics. This looks like it could be a backend issue on our side. Would you be able to share your full training script(s) that you used for this project? That would be immensely useful for our debugging.

Thanks in advance for your help, and sorry again for dropping off! I should have gotten back to you before the holidays.

**Comment by laxels** [comment]
@andreimargeloiu I'm currently investigating the issue and would like to inquire: Did you use `wandb.Api()` to update your run while it was running? If that's the case, we have a known issue that is likely to cause the problems you're experiencing.

**Comment by andreimargeloiu** [comment]
@laxels, thank you for trying to help out.

1) *Regarding my use-case*: I don't think your solution would actually work... In pseudo-code, I'm trying to:
```
validation_accuracy = [array with validation accuracy at every epoch]
validation_loss = [array with validation loss at every epoch]
print(validation_accuracy[np.argmin(validation_loss)])
```

For example, I want:
```
validation_loss = [0.5, 0.4, 0.45]
validation_accuracy = [0.7, 0.8, 0.9]

print(validation_accuracy[np.argmin(validation_loss)])  # save 0.7 in wandb
```

**Comment by andreimargeloiu** [comment]
2) *Regarding issue*: I am logging everything using pytorch-lightening function `self.log`

Specifically, my model has:
- an attribute `self.log_test_key`, which is the wandb metric key I want to log
- call ``` self.log(f'{self.log_test_key}/balanced_accuracy', balanced_accuracy_score(y_true, y_pred)) ``` from within `def validation_step()`

To compute these metrics and log to wandb I do:
```
model.log_test_key = 'bestmodel_valid'
trainer.test(model, dataloaders=data_module.val_dataloader(), ckpt_path=checkpoint_callback.best_model_path)
```

**Comment by laxels** [comment]
> @laxels, thank you for trying to help out.
> 
> 1. _Regarding my use-case_: I don't think your solution would actually work... In pseudo-code, I'm trying to:
> 
> ```
> validation_accuracy = [array with validation accuracy at every epoch]
> validation_loss = [array with validation loss at every epoch]
> print(validation_accuracy[np.argmin(validation_loss)])
> ```
> 
> For example, I want:
> 
> ```
> validation_loss = [0.5, 0.4, 0.45]
> validation_accuracy = [0.7, 0.8, 0.9]
> 
> print(validation_accuracy[np.argmin(validation_loss)])  # save 0.8 in wandb
> ```

I see. So you want the accuracy for the step with the lowest loss. Unfortunately, we don't have a simpler solution for that use case at the moment.

We're looking into what's causing your values to not be logged properly. Would it be possible for you to share your full training script (with potentially sensitive data removed)? That would be immensely helpful for reproducing the bug on our end.

**Comment by andreimargeloiu** [comment]
Sorry, I don't have the script at the point in time. I don't seem to experience the bug anymore, but I'm sure the bug is still there.
